support-frontend/assets/components/thankYou/marketingConsent/marketingConsentItems.tsx

```diff
@@ -119,29 +118,33 @@ function ThankYouMarketingConsentCTA({
 	email,
 	csrf,
 	marketingConsentState,
-	setMarketingConsentState,
 }: {
 	email: string;
 	csrf: CsrfState;
 } & ThankYouMarketingConsentState): JSX.Element {
-	const subscribeToNewsLetter = (email: string, csrf: CsrfState) => {
-		const dispatch = useContributionsDispatch();
+	const dispatch = useContributionsDispatch();
 
+	const subscribeToNewsLetter = (email: string, csrf: CsrfState) => {
 		sendMarketingPreferencesToIdentity(true, email, dispatch, csrf);
 	};
 
 	const onSubmit = () => {
 		if (!marketingConsentState.hasConsented) {
-			setMarketingConsentState((state) => ({
-				...state,
-				errorMessage: ERROR_MESSAGE,
-			}));
+			dispatch(
+				setThankYouMarketingConsent({
+					...marketingConsentState,
+					errorMessage: ERROR_MESSAGE,
+				}),
+			);
 		} else {
+			dispatch(
+				setThankYouMarketingConsent({
+					...marketingConsentState,
+					hasBeenCompleted: true,
+				}),
+			);
+			if (window.location.hostname !== 'theguardian.com') return;
```

Comment:
What's the reason for this new `window.location.hostname` conditional?

Reply:
Ah these were put in to prevent the function being called when it's not on prod